src/ensure.stanza

```diff
@@ -133,4 +133,8 @@ public defn ensure-in-set! (accepted:Tuple<Int>):
   ):
     if not contains?(accepted, value):
       throw $ ValueError("Value '%_' not in Accepted Value Set: [%,]" % [value, accepted])
-  ensure-func
\ No newline at end of file
+  ensure-func
+
+public defn ensure-not-empty! (field:String, value:Collection) :
+  if count(value) == 0:
+    throw $ ValueError("Expected Non-Empty Collection - Found: %_" % [value])
```

Comment:
What does the argument `field:String` do here?

Reply:
This is the interface for the `ensure => FUNC` members of a `defstruct` field. I believe it gets used by the `defstruct` macro to tell the user which field of the defstruct had this fault case.